lms/static/scripts/postmessage_json_rpc/server/methods.js

```diff
@@ -14,3 +14,24 @@ export function requestConfig() {
   const clientConfigObj = JSON.parse(configEl.textContent).hypothesisClient;
   return clientConfigObj;
 }
+
+/**
+ * Return the groups for the Hypothesis client to show.
+ */
+export async function requestGroups() {
+  const configObj = JSON.parse(
+    document.querySelector('.js-config').textContent
+  );
+
+  function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  if (configObj.dev === true) {
+    // Artifically sleep to simulate how long this will take when it needs to
+    // send real API requests to get the groups.
+    await sleep(1900);
+  }
+
+  return configObj.hypothesisClient.services[0].groups;
```

Comment:
This is hacky of course (reaching into the client config object to extract the groups) but I think that's fine because it's temporary. We're working as fast as possible to replace this with something else entirely (getting the groups from the sync API response), so I don't think it's worth doing anything like redesigning the JavaScript config to better support this temporary hack in the meantime.